breadbox/breadbox/service/metadata.py

```diff
@@ -162,6 +162,28 @@ def get_labels_for_slice_type(
         )
 
 
+def get_dataset_feature_label_by_id(
+    db: SessionWithUser, dataset: MatrixDataset, given_id: str
+):
+    """Looks up a single label for a given_id in a dataset. Internally this is implemented inefficiently so only
+    call this if you really want a single label. If you want to loop through many IDs, use get_matrix_dataset_sample_labels_by_id instead
+    (or we rewrite this function to make a DB query to fetch only the data we actually need)
+    """
```

Comment:
Excellent! I see what you mean now about this making it easier for us to improve performance later on.